src/hotspot/share/oops/oop.hpp

```diff
@@ -247,15 +247,17 @@ class oopDesc {
   void verify_forwardee(oop forwardee) NOT_DEBUG_RETURN;
 
   inline void forward_to(oop p);
-  inline bool cas_forward_to(oop p, markWord compare, atomic_memory_order order = memory_order_conservative);
+  inline void forward_to_self();
```

Comment:
Yes, this rename makes total sense for upstream. Want to do it?

Reply:
This is not the rename. cas_forward_to() is only used in a single place for self-forwarding, but the replacement in upstream would be forward_to_atomic(), which is doing almost exactly the same (except it returns a value not a bool). Filed:
https://bugs.openjdk.java.net/browse/JDK-8270041